lib/lightning/invocation.ex

```diff
@@ -650,4 +652,59 @@ defmodule Lightning.Invocation do
     do:
       logs_for_step(step)
       |> Enum.map_join("\n", fn log -> log.message end)
+
+  @doc """
+  Exports work orders by performing a series of database operations wrapped in a transaction.
+
+  This function creates an audit log, a project file record, and enqueues an export job using a transaction.
+  Each step is executed as part of an `Ecto.Multi` operation, ensuring atomicity.
+
+  ## Parameters
+
+    - `project` - The project for which work orders are being exported. Expected to be a map or struct with an `id` field.
+    - `user` - The user initiating the export operation. Expected to be a map or struct with an `id` field.
+    - `search_params` - A map of search parameters used to filter work orders to export.
+
+  ## Returns
+
+    - `{:ok, %{audit: audit, project_file: project_file, export_job: job}}` on success:
+      - `audit`: The audit log entry created for the export operation.
+      - `project_file`: The project file record created for the export operation.
+      - `export_job`: The result of the export job enqueuing step.
+
+    - `{:error, step, reason, changes}` if any step in the transaction fails:
+      - `step`: The step where the error occurred, e.g., `:audit`, `:project_file`, or `:export_job`.
+      - `reason`: The reason for the failure, typically an error atom.
+      - `changes`: A map of changes up to the point of failure.
+  """
+  def export_workorders(project, user, search_params) do
+    Ecto.Multi.new()
+    |> Ecto.Multi.run(:audit, fn _repo, _changes ->
+      ExportAudit.event(
+        "started",
+        project.id,
+        user.id,
+        %{},
+        %{search_params: search_params}
+      )
+      |> ExportAudit.save()
```

Comment:
@elias-ba This is interesting - in my work with the snapshot events I have (IIRC) been doing a `Multi.insert` and then returning the changeset from the `event` call.  

Would you mind sharing any advantages that you think can be found when using `Multi.run` with `ExportAudit.save`, please? (Asking for a friend :)).

Reply:
I think this is just me missing the high value of using multis. I should use multi insert, thanks for the catch I will update it. My implementation is redundant